core/src/main/java/com/graphhopper/routing/ch/NodeBasedWitnessPathSearcher.java

```diff
@@ -30,41 +29,25 @@
  * i.e. its a Dijkstra search that allows re-using the shortest path tree for different searches with the same origin
  * node and uses large int/double arrays instead of hash maps to store the shortest path tree (higher memory consumption,
  * but faster query times -> better for CH preparation). Main reason we use this instead of {@link DijkstraOneToMany}
- * is that we can use this implementation with a {@link PrepareCHGraph}.
+ * is that we can use this implementation with a {@link CHPreparationGraph} and we are only interested in checking for
+ * witness paths (e.g. we do not need to find the actual path).
  */
 public class NodeBasedWitnessPathSearcher {
-    private static final int EMPTY_PARENT = -1;
     private static final int NOT_FOUND = -1;
-    private final PrepareCHGraph graph;
-    private final PrepareCHEdgeExplorer outEdgeExplorer;
+    private PrepareGraphEdgeExplorer outEdgeExplorer;
     private final IntArrayList changedNodes;
-    private final int maxLevel;
     private int maxVisitedNodes = Integer.MAX_VALUE;
     protected double[] weights;
-    private int[] parents;
-    private int[] edgeIds;
```

Comment:
No need for these two arrays as we never need to find the actual shortest path (we just need to check its existence). This saved 2*4byte*N where N=number of nodes of the graph